master/buildbot/test/unit/test_step_artifact.py

```diff
@@ -314,3 +313,121 @@ def test_download_artifact_reusing_build(self):
+        br3 = fakedb.BuildRequest(id=3, buildsetid=3, buildername="B", triggeredbybrid=1, artifactbrid=666)
+        br4 = fakedb.BuildRequest(id=4, buildsetid=4, buildername="B", triggeredbybrid=1, artifactbrid=666)
+        br666 = fakedb.BuildRequest(id=666, buildsetid=5, buildername="B", triggeredbybrid=None)
+
+        self.setupStep(
+            artifact.DownloadArtifactsFromChildren(
+                workdir='build',
+                artifactServer='usr@srv.com',
+                artifactServerDir='/artifacts',
+                artifactServerPort=22,
+                artifactDirectory='mydir',
+                artifactBuilderName='B',
+                artifactDestination='./base/local',
+        ), [br2, br3, br4, br666])
+
+        expectedRemote1 = '\'usr@srv.com:/artifacts/B_2_01_01_1970_00_00_00_+0000/\''
+        expectedRemote2 = '\'usr@srv.com:/artifacts/B_666_01_01_1970_00_00_00_+0000/\''
+
+        expectedLocal1 = './base/local/2'
+        expectedLocal2 = './base/local/666'
+
+        self.expectCommands(
+            ExpectShell(workdir='build', usePTY='slave-config',
+                        command=['mkdir', '-p', expectedLocal1]) +
+            0,
+            ExpectShell(workdir='build', usePTY='slave-config',
+                        command='for i in 1 2 3 4 5; do rsync -var --progress --partial ' +
+                                expectedRemote1 + " '" + expectedLocal1 + "'"
+                                ' --rsh=\'ssh -p 22\'; if [ $? -eq 0 ]; then exit 0; else sleep 5; fi; done; exit -1'
+                        ) +
+            0,
+            ExpectShell(workdir='build', usePTY='slave-config',
+                        command=['mkdir', '-p', expectedLocal2]) +
+            0,
+            ExpectShell(workdir='build', usePTY='slave-config',
+                        command='for i in 1 2 3 4 5; do rsync -var --progress --partial ' +
+                                expectedRemote2 + " '" + expectedLocal2 + "'" +
+                                ' --rsh=\'ssh -p 22\'; if [ $? -eq 0 ]; then exit 0; else sleep 5; fi; done; exit -1'
+                        ) +
+            0 +
+            ExpectShell.log('stdio', stdout='')
+        )
+
+        self.expectOutcome(result=SUCCESS, status_text=['Downloaded artifacts from 2 partitions'])
+        return self.runStep()
+
+    def test_download_artifact_fromchildren_windows(self):
+        br2 = fakedb.BuildRequest(id=2, buildsetid=2, buildername="B", triggeredbybrid=1)
+
+        self.setupStep(
+            artifact.DownloadArtifactsFromChildren(
+                workdir='build',
+                artifactServer='usr@srv.com',
+                artifactServerDir='/artifacts',
+                artifactServerPort=22,
+                artifactDirectory='mydir',
+                artifactBuilderName='B',
+                artifactDestination='./base/local'
+        ), [br2], winslave=True)
+
+        expectedRemote = '\'usr@srv.com:/artifacts/B_2_01_01_1970_00_00_00_+0000/\''
```

Comment:
This cannot be correct? the `artifactDirectory` is defined to `mydir` so the expected remote should be `usr@srv.com:/artifacts/B_2_01_01_1970_00_00_00_+0000/mydir` right?